### PR TITLE
Adding Feature API documentation

### PR DESCRIPTION
**Pull request by Laimiux**
## What
- Renaming `Integration.md` to `Formula-Android.md`
- Updating documentation that now replaces `Integration` and `FragmentContract` with `FragmentKey` and `FeatureFactory`.

### CODE REVIEW

File: docs/Formula-Android.md [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+                store(MyActivityComponent(this)) {
+                    bind(CounterFeatureFactory())
+                }
+            }
+        }
+    }
+}
+```
+
+## Handling back button events
```
Comment (Laimiux) [inline docs/Formula-Android.md:220]: This section didn't change.

File: docs/Formula-Android.md [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+     override fun onBackPressed() {
+         if (!FormulaAndroid.onBackPressed(this)) {
+             super.onBackPressed()
+         }
+     }
+}
+```
+This is already in place for you if you use `FormulaAppCompatActivity`.
+
+## Fragment flow
```
Comment (Laimiux) [inline docs/Formula-Android.md:257]: I've only made small changes here, will leave this for later to think about.

**Comment by carrotkite** [comment]
### JaCoCO Code Coverage 76.49% :white_check_mark:
| Class | Covered | Meta | Status |
|:---|:---:|:---:|:---:|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: docs/Formula-Android.md [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+    - Ability to group multiple fragments into a flow and share state between them.
+    - Type-safe and scoped fragment event handling. (Avoid casting activity to a listener)
+
+## Using Fragments
+This module provides an API to connect state management and view rendering logic to Android 
+fragments. For this example, we will connect `CounterRenderView` and `CounterFormula` from the 
+main getting started [guide](index.md). 
+
+### Define a fragment key
+Fragment key is used an a unique identifier for a `FormulaFragment`.
```
Comment (Jawnnypoo) [inline docs/Formula-Android.md:18]: How unique does it have to be? What if I use the same fragment key twice?

File: docs/Formula-Android.md [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+    override val tag: String = "counter"
+) : FragmentKey
+```
+
+### Define a feature factory
+Feature factory creates state observable and a view factory for a fragment with a 
+specific `FragmentKey` type. To continue our example, we define a `CounterFeatureFactory` 
+which will handle `CounterKey` fragments.
+```kotlin
+class CounterFeatureFactory : FeatureFactory<Any, CounterKey> {
```
Comment (Jawnnypoo) [inline docs/Formula-Android.md:37]: Could this `Any` also be `Unit` if we don't have any dependencies?

File: docs/Formula-Android.md [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+        }
+    }
+}
+```
+
+If your `Activity` has another base class, you can just copy logic from `FormulaAppCompatActivity` into your `Activity`.
+
+### And that's it
+Formula takes care of the rest. The RxJava state stream is instantiated and subscribed to when the user enters 
+declared navigation destination. We dispose of the stream only when user exits the destination.
```
Comment (Jawnnypoo) [inline docs/Formula-Android.md:103]: Maybe a mention of what happens if the screen were to be rotated or another configuration change happens.

**Review by Jawnnypoo — APPROVED** [review]

File: docs/Formula-Android.md (hunk 4 above)
Comment (Laimiux) [reply docs/Formula-Android.md:37]: Currently no, but maybe a good idea to add support for that.

**Review by johnwilde — APPROVED** [review]
Just a few nits for readability.

File: docs/Formula-Android.md (hunk 3 above)
Comment (Laimiux) [reply docs/Formula-Android.md:18]: I'm updating this to the following

> Fragment key is used to instantiate `FormulaFragment` and to identify which `FeatureFactory` to 
use. You can also use it to add arguments that the fragment instance needs.

File: docs/Formula-Android.md (hunk 5 above)
Comment (Laimiux) [reply docs/Formula-Android.md:103]: Updated to 
> Formula takes care of the rest. This is how the state observable works:
> - When FormulaFragment is added we instantiate and subscribe to the state observable.
> - When FormulaFragment is removed we destroy the state observable.
> 
> The state management observable continues to run during configuration changes or if you navigate forward.

**Pull request merged**